internal/errors.go

```diff
@@ -1,19 +1,24 @@
 package internal
 
 import (
+	"errors"
 	"io"
 	"net"
 	"strings"
 )
 
+
+var ErrClosed = errors.New("redis: client is closed")
+var ErrPoolTimeout = errors.New("redis: connection pool timeout")
+
 const Nil = RedisError("redis: nil")
 
 type RedisError string
 
 func (e RedisError) Error() string { return string(e) }
 
 func IsRetryableError(err error) bool {
-	return IsNetworkError(err)
+	return IsNetworkError(err) || err == ErrPoolTimeout || err == RedisError("ERR max number of clients reached")
```

Comment:
`err.Error() == "ERR max number of clients reached"` reads better and does not allocate.